docs/src/main/asciidoc/using-vertx.adoc

```diff
@@ -0,0 +1,483 @@
+<head>
+    <meta charset="utf-8"/>
+    <title>SSE with Vert.x - Quarkus</title>
+    <script type="application/javascript" src="streaming.js"></script>
+</head>
+<body>
+<div id="container"></div>
+</body>
+</html>
+----
+
+Our web page just has an empty `<div>` container.
+The magic, as always, lies in the Javascript code:
+
+[source,javascript]
+.META-INF/resources/streaming.js
+----
+var eventSource = new EventSource("/hello/Quarkus/streaming");
+eventSource.onmessage = function (event) {
+    var container = document.getElementById("container");
+    var paragraph = document.createElement("p");
+    paragraph.innerHTML = event.data;
+    container.appendChild(paragraph);
+};
+----
+
+IMPORTANT: Most browsers support SSE but some don't.
+More about this in  Mozilla's https://developer.mozilla.org/en-US/docs/Web/API/EventSource#Browser_compatibility[SSE browser-compatibility list].
+
+Navigate to http://localhost:8080/streaming.html.
+A new greeting should show-up every 2 seconds.
+
+[source, text]
+----
+Hello Quarkus! (Thu Mar 21 17:26:12 CET 2019)
+
+Hello Quarkus! (Thu Mar 21 17:26:14 CET 2019)
+
+Hello Quarkus! (Thu Mar 21 17:26:16 CET 2019)
+
+...
+----
+
+=== Using JSON results
+
+Vert.x API heavily relies on JSON, namely the `io.vertx.core.json.JsonObject` and `io.vertx.core.json.JsonArray` types.
+They are both supported as {project-name} web resource return types.
+
+Consider these endpoints:
+
+[source,java]
+----
+@Path("/hello")
+@Produces(MediaType.APPLICATION_JSON)
+public class VertxJsonResource {
+
+    @GET
+    @Path("{name}/object")
+    public JsonObject jsonObject(@PathParam("name") String name) {
+        return new JsonObject().put("Hello", name);
+    }
+
+    @GET
+    @Path("{name}/array")
+    public JsonArray jsonArray(@PathParam("name") String name) {
+        return new JsonArray().add("Hello").add(name);
+    }
+}
+----
+
+In your browser, navigate to http://localhost:8080/hello/Quarkus/object. You should see:
+
+[source, text]
+----
+{"Hello":"Quarkus"}
+----
+
+Then, navigate to http://localhost:8080/hello/Quarkus/array:
+
+[source, text]
+----
+["Hello","Quarkus"]
+----
+
+Needless to say, this works equally well when the JSON result is wrapped in a `CompletionStage` or a `Publisher`.
+
+== Using Vert.x Clients
+
+As you can inject a Vert.x instance, you can use Vert.x clients in a {project-name} application.
+This section gives an example with the `WebClient`.
+
+=== Picking the right dependency
+
+Depending on the API model you want to use you need to add the right dependency to your `pom.xml` file:
+
+[source, xml, subs=attributes+]
+----
+<!-- bare API -->
+<dependency>
+  <groupId>io.vertx</groupId>
+  <artifactId>vertx-web-client</artifactId>
+  <version>{vertx-version}</version>
+</dependency>
+
+<!-- Axle API -->
+<dependency>
+  <groupId>io.smallrye.reactive</groupId>
+  <artifactId>smallrye-axle-web-client</artifactId>
+  <version>{axle-version}</version>
+</dependency>
+
+<!-- RX Java 2 API -->
```

Comment:
I think you should add a few words to explain why rxjava2 does not come from a `vertx-rx-java2-web-client` module.

Reply:
Good point.